eZ/Publish/API/Repository/Tests/SetupFactory/Legacy.php

```diff
@@ -301,9 +303,18 @@ protected function getInitialData()
     protected function initializeSchema()
     {
         if (!self::$schemaInitialized) {
-            $statements = $this->getSchemaStatements();
-
-            $this->applyStatements($statements);
+            switch (self::$db) {
+                case 'pgsql':
+                    $connection = $this->getDatabaseHandler()->getConnection();
+                    $installer = new PostgreSqlCleanInstaller($connection);
+                    $installer->setOutput(new NullOutput());
+                    $installer->importSchema();
```

Comment:
Approach no.2 mentioned above would be great here (to get rid of switch), but beside that maybe using installer class here is not the greatest idea. I could create decorator for schema manager and use it both in Installer class and here.

Of course I could leave initializing schema as it was before, but I think it's valuable to test if everything is working with schema created by Doctrine.